docs/reference/openapi-rules.md

```diff
@@ -1,3 +1,546 @@
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+openapi: 3.0.0
+info:
+  version: '1.0.0'
+  title: Descriptive API
+  description: >+
+    Some description about the general point of this API, and why it exists when another similar but different API also exists.
+
+    ## Authentication
+
+    This API uses OAuth2 and tokens can be requested from [Dev Portal: Tokens](https://example.org/developers/tokens).
+```
+
+### info-license
+
+The `info` object should have a `license` key.
+
+It can be hard to pick a license, so if you don't have a lawyer around you can use [TLDRLegal](https://tldrlegal.com/) and [Choose a License](https://choosealicense.com/) to help give you an idea. 
+
+How useful this is in court is not entirely known, but having a license is better than not having a license. 
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  license:
+    name: MIT
+```
+
+### license-url
+
+Mentioning a license is only useful if people know what the license means, so add a link to the full text for those who need it.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  license:
+    name: MIT
+    url: https://www.tldrlegal.com/l/mit
+```
+
+### no-eval-in-markdown
+
+This rule protects against an edge case, for anyone bringing in description documents from third parties and using the parsed content rendered in HTML/JS. If one of those third parties does something shady like inject `eval()` JavaScript statements, it could lead to an XSS attack. 
+
+**Recommended:** No
+
+**Bad Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  title: 'some title with eval(',
+```
+
+### no-script-tags-in-markdown
+
+This rule protects against a potential hack, for anyone bringing in description documents from third parties then generating HTML documentation. If one of those third parties does something shady like inject `<script>` tags, they could easily execute arbitrary code on your domain, which if it's the same as your main application could be all sorts of terrible.
+
+**Recommended:** Yes
+
+**Bad Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  title: 'some title with <script>alert("You are Hacked");</script>',
+```
+
+### openapi-tags-alphabetical
+
+OpenAPI object should have alphabetical `tags`. This will be sorted by the `name` property.
+
+**Recommended:** No
+
+**Bad Example**
+
+```yaml
+tags: 
+  - name: 'Badger'
+  - name: 'Aardvark'
+```
+
+**Good Example**
+
+```yaml
+tags: 
+  - name: 'Aardvark'
+  - name: 'Badger'
+```
+
+### openapi-tags
+
+OpenAPI object should have non-empty `tags` array.
+
+Why? Well, you _can_ reference tags arbitrarily in operations, and definition is optional...
+
+```
+/invoices/{id}/items:
+  get:
+    tags:
+    - Invoice Items
+```
+
+Defining tags allows you to add more information like a `description`. (See [tag-description](#tag-description))
```

Comment:
```suggestion
### tag-description

Defining tags allows you to add more information like a `description`. (See [tag-description](#tag-description))
```

Reply:
Seems like wrong url?

Reply:
Huh? This sentence is "for more information check out tag-description which is another rule" its not missing a header or anything.